yaml/converter/legacy/internal/config.go

```diff
@@ -65,12 +69,53 @@ func Convert(d []byte) ([]byte, error) {
 			toContainer(container),
 		)
 	}
+
 	pipeline.Volumes = toVolumes(from)
 	pipeline.Trigger.Branch.Include = from.Branches.Include
 	pipeline.Trigger.Branch.Exclude = from.Branches.Exclude
 
-	manifest := &droneyaml.Manifest{}
-	manifest.Resources = append(manifest.Resources, pipeline)
+	if len(from.Matrix.Include) > 0 {
+		for _, environ := range from.Matrix.Include {
+			name := []string{"default"}
+
+			for _, value := range environ {
+				name = append(name, value)
+			}
+
+			current := pipeline
+			current.Name = strings.Join(name, "-")
+
+			marshaled, err := yaml.Marshal(&current)
+
+			if err != nil {
+				return nil, err
+			}
+
+			evaled, err := envsubst.Eval(string(marshaled), func(name string) string {
```

Comment:
I am not sure we want to run `envsubst` twice. Once you run `envsubst` it is going to remove all `${VARIABLE}` declarations and replace with empty strings, which could break builds that do things like `tags: ${DRONE_COMIMIT_SHA:0:6}`

Reply:
Oh, it's replacing all of them? That would be bad. So we should do the replace manually via regular expressions?

Reply:
that is probably what I would recommend. Also maybe appending the matrix variables to each step as environment variables, since some steps may be expecting these?

Reply:
OK, I have replaced it with `strings.Replace` now. If that is looking good to you I could also iterate through the steps and append the environment. Not sure if this is really required but it won't hurt.